File: pbweberogwcl/src/app/chiusura-progetto/services/chiusura-progetto.service.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

/* tslint:disable */
import { Injectable } from '@angular/core';
import { HttpClient, HttpRequest, HttpResponse, HttpHeaders } from '@angular/common/http';
import { BaseService as __BaseService } from '../../shared/base-service';
import { ApiConfiguration as __Configuration } from '../../shared/api-configuration';
import { StrictHttpResponse as __StrictHttpResponse } from '../../shared/strict-http-response'
import { Observable as __Observable } from 'rxjs';
import { map as __map, filter as __filter } from 'rxjs/operators';
import {ConfigService} from "../../core/services/config.service";
import {EsitoOperazioni} from "../../shared/api/models/esito-operazioni";
import {EsitoOperazioneChiudiProgetto} from "../commons/models/esito-operazione-chiudi-progetto";
import {RequestChiudiProgetto} from "../commons/requests/request-chiudi-progetto";


@Injectable({
  providedIn: 'root',
})
class ChiusuraProgettoService extends __BaseService {
  static readonly verificaErogazioneSaldoPath = '/chiusuraProgetto/erogazioneSaldo';
  static readonly chiudiProgettoPath = '/chiusuraProgetto/progetto';
  static readonly isRinunciaPresentePath = '/chiusuraProgetto/rinuncia';
  static readonly chiudiProgettoDiUfficioPath = '/chiusuraProgetto/ufficio';

  constructor(
    config: __Configuration,
    configService: ConfigService,
    http: HttpClient
  ) {
    super(config, configService, http);
  }

  /**
   * Verifica  se esiste il progetto associato/collegato/a contributo.
   * @param idProgetto undefined
   * @return successful operation
   */
  verificaErogazioneSaldoResponse(idProgetto?: number): __Observable<__StrictHttpResponse<EsitoOperazioni>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    if (idProgetto != null) __params = __params.set('idProgetto', idProgetto.toString());
    let req = new HttpRequest<any>(
      'GET',
      this.rootUrl + `/restfacade/chiusuraProgetto/erogazioneSaldo`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<EsitoOperazioni>;
      })
    );
  }
  /**
   * Verifica  se esiste il progetto associato/collegato/a contributo.
   * @param idProgetto undefined
   * @return successful operation
   */
  verificaErogazioneSaldo(idProgetto?: number): __Observable<EsitoOperazioni> {
    return this.verificaErogazioneSaldoResponse(idProgetto).pipe(
      __map(_r => _r.body as EsitoOperazioni)
    );
  }

  /**
   * Chiude la gestione operativa del progetto.
   * @param body undefined
   * @return successful operation
   */
  chiudiProgettoResponse(body?: RequestChiudiProgetto): __Observable<__StrictHttpResponse<EsitoOperazioneChiudiProgetto>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    __body = body;
    let req = new HttpRequest<any>(
      'POST',
      this.rootUrl + `/restfacade/chiusuraProgetto/progetto`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<EsitoOperazioneChiudiProgetto>;
      })
    );
  }
  /**
   * Chiude la gestione operativa del progetto.
   * @param body undefined
   * @return successful operation
   */
  chiudiProgetto(body?: RequestChiudiProgetto): __Observable<EsitoOperazioneChiudiProgetto> {
    return this.chiudiProgettoResponse(body).pipe(
      __map(_r => _r.body as EsitoOperazioneChiudiProgetto)
    );
  }

  /**
   * Verifica se presente una rinuncia per il progetto.
   * @param idProgetto undefined
   * @return successful operation
   */
  isRinunciaPresenteResponse(idProgetto?: number): __Observable<__StrictHttpResponse<boolean>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    if (idProgetto != null) __params = __params.set('idProgetto', idProgetto.toString());
    let req = new HttpRequest<any>(
      'GET',
      this.rootUrl + `/restfacade/chiusuraProgetto/rinuncia`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'text'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return (_r as HttpResponse<any>).clone({ body: (_r as HttpResponse<any>).body === 'true' }) as __StrictHttpResponse<boolean>
      })
    );
  }
  /**
   * Verifica se presente una rinuncia per il progetto.
   * @param idProgetto undefined
   * @return successful operation
   */
  isRinunciaPresente(idProgetto?: number): __Observable<boolean> {
    return this.isRinunciaPresenteResponse(idProgetto).pipe(
      __map(_r => _r.body as boolean)
    );
  }

  /**
   * Chiude progetto di ufficio.
   * @param body undefined
   * @return successful operation
   */
  chiudiProgettoDiUfficioResponse(body?: RequestChiudiProgetto): __Observable<__StrictHttpResponse<EsitoOperazioneChiudiProgetto>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    __body = body;
    let req = new HttpRequest<any>(
      'POST',
      this.rootUrl + `/restfacade/chiusuraProgetto/ufficio`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<EsitoOperazioneChiudiProgetto>;
      })
    );
  }
  /**
   * Chiude progetto di ufficio.
   * @param body undefined
   * @return successful operation
   */
  chiudiProgettoDiUfficio(body?: RequestChiudiProgetto): __Observable<EsitoOperazioneChiudiProgetto> {
    return this.chiudiProgettoDiUfficioResponse(body).pipe(
      __map(_r => _r.body as EsitoOperazioneChiudiProgetto)
    );
  }
}

module ChiusuraProgettoService {
}

export { ChiusuraProgettoService }

File: pbweberog/pbweberog-jar/src/main/java/it/csi/pbandi/pbweberog/business/service/ChiusuraProgettoService.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbweberog.business.service;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.QueryParam;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import it.csi.pbandi.pbservizit.exception.UtenteException;
import it.csi.pbandi.pbservizit.pbandisrv.dto.fineprogetto.EsitoOperazioneChiudiProgetto;
import it.csi.pbandi.pbweberog.dto.EsitoOperazioni;
import it.csi.pbandi.pbweberog.integration.request.RequestChiudiProgetto;

@Path("/chiusuraProgetto")
@Api(value = "chiusuraProgetto")
public interface ChiusuraProgettoService {
	
	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////	
	///////////////////////////////////////////// CHIUSURA DEL PROGETTO ///////////////////////////////////////////////////////////
	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
	
	@GET
	@Path("/rinuncia") 
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Verifica se presente una rinuncia per il progetto.", response=Boolean.class)
	Response isRinunciaPresente( @Context HttpServletRequest req, @QueryParam("idProgetto") Long idProgetto) throws UtenteException, Exception;
	
	@GET
	@Path("/erogazioneSaldo") 
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Verifica  se esiste il progetto associato/collegato/a contributo.", response=EsitoOperazioni.class)
	Response verificaErogazioneSaldo( @Context HttpServletRequest req, @QueryParam("idProgetto") Long idProgetto) throws UtenteException, Exception;
	
	
	@POST
	@Path("/progetto") 
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Chiude la gestione operativa del progetto.", response=EsitoOperazioneChiudiProgetto.class)
	Response chiudiProgetto( @Context HttpServletRequest req, RequestChiudiProgetto chiudiRequest) throws UtenteException, Exception;
	
	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////	
	///////////////////////////////////////// CHIUSURA D'UFFICIO DEL PROGETTO /////////////////////////////////////////////////////
	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
	@POST
	@Path("/ufficio") 
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Chiude progetto di ufficio.", response=EsitoOperazioneChiudiProgetto.class)
	Response chiudiProgettoDiUfficio( @Context HttpServletRequest req, RequestChiudiProgetto chiudiRequest) throws UtenteException, Exception; 
	
}
